File: isUnique.py
def isUnique(str):
    dic = {}

    for char in str:
        if not char in dic:
            dic[char] = 1
        else:
            dic[char] += 1
    
    for keys, values in dic.items():
        if values > 1:
            return False
        else:
            return True

print(isUnique('AAb'))
print(isUnique('ab'))

File: chapter1/isUnique.js
function isUnique(str) {
  let dic = {};

  for (let char of str) {
    if (char in dic) {
      return false;
    }
    dic[char] = true;
  }
  return true;
}

console.log(isUnique("abccd"));

module.exports = isUnique;
